tests/tests/loopback.rs

```diff
@@ -0,0 +1,306 @@
+use anyhow::Result;
+use opteadm::OpteAdm;
+use oxide_vpc::api::{
+    AddRouterEntryReq, Address, BoundaryServices, Direction, Filters,
+    FirewallAction, FirewallRule, IpCfg, IpCidr, Ipv4Cfg, PhysNet, Ports,
+    RouterTarget, SNat4Cfg, SetVirt2PhysReq, Vni, VpcCfg,
+};
+use std::env;
+use std::process::Command;
+use ztest::*;
+
+/// This is a wrapper around the ztest::Zone object that encapsulates common
+/// logic needed for running the OPTE tests zones used in this test suite.
+struct OpteZone<'a> {
+    _zfs: &'a Zfs,
+    zone: Zone,
+}
+
+impl<'a> OpteZone<'a> {
+    /// Create a new zone with the given name, underlying zfs instance and set
+    /// of interfaces. In illumos parlance, the interfaces are data link
+    /// devices.
+    fn new(name: &str, zfs: &'a Zfs, ifx: &[&'a str]) -> Result<Self> {
```

Comment:
Don't think you need the lifetime on the inner `str`.

```suggestion
    fn new(name: &str, zfs: &'a Zfs, ifx: &[&str]) -> Result<Self> {
```

Reply:
Fixed.